templates/azure-api-template-nodejs/components/logger.js

```diff
@@ -0,0 +1,129 @@
+/**
+    Nodejs Template Project
+    @module: logger.js
+    @description: a simple logging module for nodejs
+    @author:
+    @version: 1.0
+**/
+
+
+/*
+USAGE
+
+    logger = require('./components/logger.js')
+
+    logger.init(event, context); // by default logging level is info
+
+    logger.warn('The following waring message will be logged');
+    logger.verbose('The following waring message will not be logged')
+
+
+    // Sample logging messages.
+    logger.error('Runtime errors or unexpected conditions.');
+    logger.warn('Runtime situations that are undesirable or unexpected, but not necessarily "wrong".');
+    logger.info('Interesting runtime events (Eg. connection established, data fetched etc).');
+    logger.verbose('Generally speaking, most lines logged by your application should be written as verbose.');
+    logger.debug('Detailed information on the flow through the system.);
+
+    // Alternate usage
+    logger.log('error', 'message');
+    logger.log('info', 'message');
+
+*/
+module.exports = function () {
+    const logLevels = {
+        error: 4,
+        warn: 3,
+        info: 2,
+        verbose: 1,
+        debug: 0
+    };
+
+    const config = {
+        curLogLevel: 'info',
+        requestDetails: ''
+    };
+
+    // set logLevel, RequestDetails
+    const init = (event, context) => setLevel();
+
+    // To add request specific details, which will be prepended in all the logs for ease of debugging in CloudWatch logs
+    const setRequestDetails = (someContextSpecificId) => {
+        return;
+
+        // Timestamp and requestID are prepended in cloudwatch log by default; If any other details are required it can be done here.
+
+        /*
+        if (someContextSpecificId != undefined && someContextSpecificId != '') {
+            config.someContextSpecificId = someContextSpecificId;
+            config.requestDetails = 'someContextSpecificId : ' + someContextSpecificId + ' =>\t'
+        } else{
+            config.requestDetails = ''
+        };
+        */
+    };
+
+    // set current logLevel; Only logs which are above the curLogLevel will be logged;
+    const setLevel = (level) => {
+        // LOG_LEVEL is 'info' by default
+
+        if (level && logLevels[level]) {
+            // If LOG_LEVEL if explicitly specified , set it as the curLogLevel
+            config.curLogLevel = level;
+            return level;
+        } else {
+            // Get LOG_LEVEL from the environment variables (if defined)
+            try {
+                level = process.env.LOG_LEVEL;
```

Comment:
Would we be able to set environment variables for azure functions? If we do, can we check if it is done this way?

Reply:
For what i know from doc:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node
it looks like it is the same process.env.  But this has a carva, we are using azure log instead of console log, and azure log default level (similar as java slf4j properties file) is controlled in the host.json.  And host.json has special syntax for logging.  

`module.exports = function (context, myTimer) {
    var timeStamp = new Date().toISOString();

    context.log('Node.js timer trigger function ran!', timeStamp);
    context.log(GetEnvironmentVariable("AzureWebJobsStorage"));
    context.log(GetEnvironmentVariable("WEBSITE_SITE_NAME"));

    context.done();
};

function GetEnvironmentVariable(name)
{
    return name + ": " + process.env[name];
}`